Assets/Scripts/Pin.cs

```diff
@@ -1,98 +0,0 @@
-﻿using System;
-using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
-
-public class Pin : MonoBehaviour
-{
-
-	
-	public float PinPositionThreshold = .03f;
-	public float RotationThreshold = 3f;
-	public float PinTranslationRate = .04f;
-	public Vector3 InitialPosition;
-	public PinPosition pinPosition { get; set; }
-	
-	private Rigidbody pinRigidBody;
-	private AudioSource audioSource;
-
-	
-	
-	private void Start()
-	{
-		audioSource = gameObject.GetComponent<AudioSource>();
-		pinRigidBody = gameObject.GetComponent<Rigidbody>();
-	}
-	
-	public bool PinAtPinPosition()
-	{
-		var position = pinPosition.GetPosition();
-		var pinCurrentPosition = transform.position;
-		return (Mathf.Abs(position.x - pinCurrentPosition.x) <= PinPositionThreshold &&
-		        Mathf.Abs(position.z - pinCurrentPosition.z) <= PinPositionThreshold);
-	}
-
-	public bool IsStanding()
```

Comment:
There is a much easier way to decide if the pin is standing up. Come chat if this needs some explaining ;)

```
const float PinStandingThreshold = 0.7f;
var isStandingUp = Vector3.Dot(transform.up, Vector3.up) >= PinStandingThreshold
```

The dot product we're using here, gives the amount of the first vector in the direction of the second vector. As both vectors are unit length, this means it literally gives you the percentage (0.0f would mean it's perpendicular, 1.0f would mean it's exactly parallel). 

0.7f is ~ the Cosine of 45 degrees which is the point at which the pin is pointing more up than sideways or down. If you increase the value the pin needs to be more vertical to be considered standing.